resources/js/mp-history/score.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import FlagCountry from 'components/flag-country';
+import Mod from 'components/mod';
+import Ruleset from 'interfaces/ruleset';
+import ScoreJson, { ScoreStatisticsAttribute } from 'interfaces/score-json';
+import UserJson from 'interfaces/user-json';
+import { route } from 'laroute';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import { formatNumber } from 'utils/html';
+import { trans } from 'utils/lang';
+
+interface Props {
+  mode: Ruleset;
+  score: ScoreJson;
+  users: Partial<Record<number, UserJson>>;
+}
+
+const firstRow = ['combo', 'accuracy', 'score'];
+const secondRow: ScoreStatisticsAttribute[] = ['count_geki', 'count_300', 'count_katu', 'count_100', 'count_50', 'count_miss'];
+
+export default function Score(props: Props) {
+  const user = props.users[props.score.user_id];
+
+  if (user == null) {
+    return null;
+  }
```

Comment:
Not yet sure what the accepted solution is for handling "impossible" situations like these. Previous code didn't care about this missing, so maybe it's fine?

Note that a similar situation happens below with `props.score.match` being possibly undefined twice, but in those cases there are *somewhat sane* fallbacks available, so I just handled locally. Maybe a `props.score.match == null` check should be here too.

Reply:
`throw new Error`